lib/game.rb

```diff
@@ -0,0 +1,124 @@
+require 'pry'
+class Game
+attr_accessor :board, :player_1, :player_2
+
+    def initialize(player_1= Players::Human.new("X"), player_2=Players::Human.new("O"), board=Board.new)
+      @board=board || Array.new(9, " ")
+      @player_1=player_1
+      @player_2=player_2
+    end #end of initialize
+
+   WIN_COMBINATIONS=[
+     [0,1,2], #top horizontal win
+     [3,4,5], #Middle horizontal win
+     [6,7,8], #Bottomr horizontal win
+     [0,3,6], #Left Vert win
+     [1,4,7], #Middle Vert win
+     [2,5,8], #Right Vert Win
+     [0,4,8], #Diagonal win
+     [2,4,6] #Diagonal win
+     ]
+
+     def current_player
+        @board.turn_count.even? ? @player_1 : @player_2
+      end #end of current_player
+
+def over?
+won? || draw?
+end #end of over?
+
+def draw?(board)
+(full?(board) && !won?(board))
+end #end of draw?
+
+
+def won?
```

Comment:
Instead of calling `Game#won?` twice, consider using an if statement assignment. For example:

```ruby
if won = won?
  board.cells[won.first]
end
```

If `won?` returns true, then and only then will the local variable `won` be assigned, otherwise, the method will just abort with a false return value. But if `won` is assigned, we can use that variable instead of calling `won?` again within the if statement.